src/zkevm_specs/evm/execution/oog_constant.py

```diff
@@ -0,0 +1,44 @@
+from ...util import FQ
+from ..instruction import Instruction, Transition
+from ..table import CallContextFieldTag
+from ..execution_state import ExecutionState
+from ..opcode import Opcode
+from ...util import N_BYTES_GAS
+
+
+def oog_constant(instruction: Instruction):
+    # retrieve op code associated to oog constant error
+    opcode = instruction.curr.aux_data
```

Comment:
Maybe the opcode should be retrieved via constraints like:
```
opcode = instruction.opcode_lookup(True)
```
So that the opcode value is guaranteed to be the one in the step.

Reply:
I tried this way before, it failed. for example, byte code of `PUSH1 0x80` when do `opcode_lookup` the program counter offset come to `0x80` location, which is not code , `opcode_lookup` only works for non `PUSH*` op codes.

Reply:
oh, I realize it is caused by incorrect pc setting in the test, updated now in [ca99681](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/266/commits/ca99681bdfc4867b4ee10d2dd5c1d8170d282121) thanks for catching !